base/sparse/sparsematrix.jl

```diff
@@ -173,7 +173,9 @@ end
 function sparse_compute_reshaped_colptr_and_rowval{Ti}(colptrS::Vector{Ti}, rowvalS::Vector{Ti}, mS::Int, nS::Int, colptrA::Vector{Ti}, rowvalA::Vector{Ti}, mA::Int, nA::Int)
     lrowvalA = length(rowvalA)
     maxrowvalA = (lrowvalA > 0) ? maximum(rowvalA) : zero(Ti)
-    if length(colptrA) != (nA+1) || maximum(colptrA) > lrowvalA+1
+    if length(colptrA) != nA+1
+        throw(DimensionMismatch("length of colptrA must be $(nA+1) but is $(length(colptrA))"))
+    elseif maximum(colptrA) > lrowvalA+1
         Base.throw_boundsError(A,colptrA)
     elseif maxrowvalA > mA
         Base.throw_boundsError(A,maxrowvalA)
```

Comment:
Seems this method receives only `A`'s structure by way of `colptrA`/`rowvalA`/`mA`/`nA` rather than `A` itself? So that branch must not be tested, as otherwise tests would fail? Best!